src/Juvix/Core/IR/Typechecker.hs

```diff
@@ -214,3 +214,26 @@ typeElim p ii ctx elim@(IR.Ann π theTerm theType level) = do
   let ann = Annotation π ty
   theTerm' <- typeTerm p ii ctx theTerm ann
   pure $ Typed.Ann π theTerm' theType' level ann
+
+
+-- | Subtyping. If @s <: t@ then @s@ is a subtype of @t@, i.e. everything of
+-- type @s@ can also be checked against type @t@.
+--
+-- Currently subtyping consists of the following:
+--
+-- * Consistency of universe levels (@*ᵢ <: *ⱼ@ if @i ≤ j@)
+-- * Usage compatibility (@(π x: A) → B <: (ω x: A) → B@ for finite @π@)
+-- * Contravariant domain & covariant codomain
+--   (@(π x: A₁) → B₁ <: (π x: A₂) → B₂@ if
+--    @A₂ <: A₁@ and @B₁ <: B₂@)
+-- * It doesn't descend into any other structures
+--   (TODO: which ones are safe to do so?)
```

Comment:
I'm not sure off the top of my head, we should formalise this eventually though - https://github.com/cryptiumlabs/juvix/issues/377.